src/test/java/com/google/cloud/spanner/pgadapter/wireoutput/ErrorResponseTest.java

```diff
@@ -102,4 +157,63 @@ public void testSendPayloadWithHint() throws Exception {
         "Length: 51, Error Message: test message, Hints: test hint\n" + "line 2",
         response.getPayloadString());
   }
+
+  @Test
+  public void testSendPayloadWithClientHint() throws Exception {
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    DataOutputStream output = new DataOutputStream(out);
+    ConnectionMetadata metadata = mock(ConnectionMetadata.class);
+    when(metadata.getOutputStream()).thenReturn(output);
+    ConnectionHandler connectionHandler = mock(ConnectionHandler.class);
+    when(connectionHandler.getWellKnownClient()).thenReturn(WellKnownClient.PGBENCH);
+    when(connectionHandler.getConnectionMetadata()).thenReturn(metadata);
+    ErrorResponse response =
+        new ErrorResponse(
+            connectionHandler,
+            PGException.newBuilder("test message")
+                .setSeverity(Severity.ERROR)
+                .setSQLState(SQLState.RaiseException)
+                .build());
+
+    response.sendPayload();
+
+    assertEquals(
+        String.format(
+            "SERROR\0CP0001\0Mtest message\0H%s\0\0", ClientAutoDetector.PGBENCH_USAGE_HINT),
```

Comment:
As I understand, this test case includes a few nulls (represented by the `\0`), right? If not could we test that case as well?

Reply:
The `\0` are not nulls, but markers in the PG wire-protocol that the specific part of the message ends at that position. So instead of sending for example the length of a string, the wire-protocol uses null bytes to indicate the end of a field.